src/index.js

```diff
@@ -9,15 +9,18 @@ module.exports = function gatsbyRemarkCodeTitles(
     const [language, ...params] = (node.lang || '').split(':');
     const query = params.join('&');
     const options = qs.parse(query);
-    const title = options.title;
+    const { title, ...rest } = options;
     if (!title || !language) {
       return;
     }
 
-    delete options['title'];
     let newQuery = '';
-    for (let key in options) {
-      newQuery += `:${key}=${options[key]}`;
+    if (Object.keys(rest).length) {
```

Comment:
rest will always be an object, and the empty case will be handled already. Not needed.